frontend/vue.config.js

```diff
@@ -28,6 +30,17 @@ module.exports = {
       .end()
       .use('vue-svg-loader')
       .loader('vue-svg-loader')
+
+    config.module
+      .rule('source-map-loader')
+      .test(/\.(js)$/)
+      .enforce('pre')
+      .include
+      .add(path.resolve(__dirname, 'node_modules/hal-json-vuex'))
+      .end()
+      .use('source-map-loader')
+      .loader('source-map-loader')
+      .end()
```

Comment:
I don't exactly understand all the lines here. Here are my assumptions:
* `rule` is for starting and naming a new rule
* `test`, `use`, `enforce` and `include.add` are copied and adapted from the article you linked
* `end` ends the rule specification
Why is is necessary to write the `use`, `loader`, `end` lines?

Reply:
To be honest, I don't understand understand it fully either. It's just copy-pasta from different sources and it works ... I used `vue inspect` to check that we get the desired webpack config, as mentioned here https://cli.vuejs.org/guide/webpack.html#chaining-advanced. You'd need to dig into the docs of https://github.com/neutrinojs/webpack-chain to understand what's going on exactly.